.github/workflows/tic.yml

```diff
@@ -18,15 +18,16 @@ jobs:
   all:
     runs-on: ${{ matrix.config.os }}
 
-    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+    name: ${{ matrix.config.os }} - r-${{ matrix.config.r }} - mlr3-${{ matrix.config.mlr3 }}
```

Comment:
I don't think you need to change the matrix naming and cache, conditioning on an env var as in the `config:` and then adding the if-condition as you did in `tic.R` should be enough.